src/sonic-yang-models/yang-models/sonic-bgp-neighbor.yang

```diff
@@ -0,0 +1,107 @@
+module sonic-bgp-neighbor {
+    namespace "http://github.com/Azure/sonic-bgp-neighbor";
+    prefix sbnbr;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-bgp-common {
+        prefix sbc;
+    }
+
+    import sonic-port {
+        prefix prt;
+    }
+
+    import sonic-portchannel {
+        prefix spc;
+    }
+
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+    import sonic-bgp-global {
+        prefix sbgpg;
+    }
+
+    import sonic-bgp-peergroup {
+        prefix pg;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC BGP Neighbor";
+
+    revision 2021-02-26 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-bgp-neighbor {
+        container BGP_NEIGHBOR {
+            list BGP_NEIGHBOR_LIST {
+                key "vrf_name neighbor";
+
+                leaf vrf_name {
+                    type leafref {
+                        path "/sbgpg:sonic-bgp-global/sbgpg:BGP_GLOBALS/sbgpg:BGP_GLOBALS_LIST/sbgpg:vrf_name";
+                    }
+                    description "Network-instance/VRF name";
+                }
+
+                leaf neighbor {
+                    type union {
+                        type inet:ip-address;
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:port_name";
+                        }
+                        type leafref {
+                            path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:portchannel_name";
```

Comment:
@venkatmahalingam : This PR (https://github.com/Azure/sonic-buildimage/pull/7049) is merged. Shall we please update the interface leafref's names accordingly.

Reply:
done.